api/handler/auth/login.go

```diff
@@ -36,6 +34,23 @@ func Login(userStore store.UserStore) http.HandlerFunc {
 			Password: request.Password,
 		}
 
+		state, err := userStore.GetUserState(ctx, newLogin)
```

Comment:
Yes, probably better to have `EmailActivated` with boolean return. I usually don't prefer state driven by integers cause it might get easily abused for various edge cases and the possible integer grows uncontrollably.

TL;DR; FYI, There might be cases where it is beneficial though
Probably when we already know all the states at start and no of the state intersect (e.g. P, R, S are different states, there is no such cases when when P and R can happen simultaneously which results in S)
Another way to create a more sophisticated integer-based state is by aggregate/summing binaries of an integer. e.g. what they do in `chmod` command.
Anyway, in code, each integer state must be represented by enum or constant